lib/types/url.js

```diff
@@ -2,10 +2,11 @@ const url = require('url');
 const normalize = require('../normalize');
 const ensureString = require('../ensureString');
 
-const isValidUrl = uri => (uri.protocol != null) &&
-uri.protocol.match(/^http[s]?:/) &&
-uri.slashes &&
-uri.hostname;
+const isValidUrl = function (uri) {
+  return (uri.protocol != null) &&
+    uri.protocol.match(/^http[s]?:/) &&
```

Comment:
```suggestion
    /^http[s]?:/.test(uri.protocol) &&
```
Since we are not doing anything with the match and simply want a boolean, it would be more effective to use `.test` method.